### PR TITLE
refactor: use graphql helper for all graphql calls

### PR DESCRIPTION
**Pull request by wa0x6e**
This refactor is using the `graphQlCall` helper for all graphql calls

This will enable us in #265 to put the api key handling only inside this function

### CODE REVIEW

**Review by ChaituVR — APPROVED** [review]
utAck

File: src/resolvers/space-sx.ts [hunk 1]
```diff
@@ -21,25 +20,22 @@ async function getSpaceProperty(key: string, url: string, property: 'avatar' | '
...
-        }`
-    },
-    ...axiosDefaultParams
-  });
-
-  const result = data.data?.data?.spaces
-    ?.map(space => space.metadata?.[property])
-    .filter(Boolean)[0];
+      data: { spaces }
+    }
```
Comment (ChaituVR) [inline src/resolvers/space-sx.ts:26]: should be fine in case of an error?

Comment (wa0x6e) [reply src/resolvers/space-sx.ts:26]: Mhhh it’s been tested only in case of no results

Comment (ChaituVR) [reply src/resolvers/space-sx.ts:26]: For now, i think we should go back to the previous way `data.data?.data?.spaces` and we should be good to merge

Comment (ChaituVR) [reply src/resolvers/space-sx.ts:26]: > Also, constructing the queries like this is a bit error prone:

What kind of errors you are referring to?

**Comment by ChaituVR** [comment]
> @wa0x6e @ChaituVR @bonustrack In favor of moving forward, I will create an issue for the API response validation and GraphQL variables. Let's make this here just work. It does, right?

Yep let's go with this for now. it looks like API response validation doesn't work before as well

**Pull request merged**